awx/ui/client/lib/models/Credential.js

```diff
@@ -40,22 +45,73 @@ function assignInputGroupValues (inputs) {
     });
 }
 
+function setDependentResources (id) {
+    this.dependentResources = [
+        {
+            model: new ProjectModel(),
```

Comment:
I think it'd be nice to be able to pass these `params` straight to the constructor.